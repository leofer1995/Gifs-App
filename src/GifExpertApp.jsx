import React, { useState } from "react";
import AddCategory from "./Components/AddCategory";
import GifGrid from "./Components/GifGrid";

const GifExpertApp=()=>{

    const [categories,setCategories]= useState([])
    
    // const handleAdd = ()=>{
    //     setCategories([...categories, 'Los Simpsons'])
    // }
    
    return (
        <>
            <h2>GiffiApp</h2>
            <AddCategory setCategories={setCategories}/>
            <hr />
            {/* <button onClick={handleAdd}>Agregar</button> */}
            <ol>
                {categories.map((category,i)=>
                    <GifGrid 
                        key = {i}
                        category={category}
                    />)}
            </ol>
        </>
    )
}

export default GifExpertApp;