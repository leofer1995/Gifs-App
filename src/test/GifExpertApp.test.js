import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp"


describe('pruebas en <GifExpertApp />',()=>{
    test('deberia mostrarse correctamente',()=>{
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostar una lista de categorias',()=>{
        const categories = ['Rick and Morty', 'One Punch'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
        
    })
})