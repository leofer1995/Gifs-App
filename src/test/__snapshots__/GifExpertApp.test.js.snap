// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifExpertApp /> deberia mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GiffiApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
